### ISSUE TITLE
Create a test suite for servers of triple pattern fragments

### ISSUE THREAD

**Issue by RubenVerborgh** [HydraCG/Specifications#65]
Write a utility that verifies whether a server obeys the [triple pattern fragments spec](http://www.hydra-cg.com/spec/latest/triple-pattern-fragments/).

**Comment by Mec-iS** [comment]
I think this can be updated by renaming it "Creating a test suite for servers to comply with Hydra specs" that should imply creating automated tests to check if a server is compliant with the use cases. Every use case can be defined into code and all the tests collected in a test battery.
Let me know if this sounds right to you, we may assign this task to a GSOC students in 2019.

**Comment by tpluscode** [comment]
I never really grasped the tight relation between TPF and Hydra Community Group.

> I think this can be updated by renaming it 

I'd rather create a new issue. That said, such a test suite has been proposed already at some point (or multiple times). Personally I'm at a loss trying to imagine how such a test suite would be executed and correctness verified. 

Please, if you have ideas for its shape please create a ticket with some discussion

**Comment by Mec-iS** [comment]
I created a task https://github.com/HTTP-APIs/python-hydra-agent/issues/61 to discuss the implementation. Hope students will get interested in it. I think that is the right place for an implementation, as this repository is mostly used for vocabulary .

The idea is to use the POC we created this year (`python-hydra-agent`) to perform step-by-step test cases that trace the procedures mentioned in the specs use cases, so to provide a battery of Python unit tests that interrogates the target server to check for proper behaviour in the request-response flow. This will improve both our knowledge of practical Hydra usage and our server's implementation.

At this point we have just a generic statement of task that will evolve into a "GSOC Project Idea" that can collect different of such tasks. Any contribution of ideas welcome, the more the tasks are detailed, the more the probability of Google to accept our project.
